### PR TITLE
Allow PostgreSQL database connection using Unix socket

### PR DESCRIPTION
**Pull request by yann-soubeyrand**
PostgreSQL database connection allows authentication based on system users (peer method) which is convenient when using PHP FPM with a dedicated user.

### CODE REVIEW

File: lib/PicoDb/Driver/Postgres.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Postgres extends Base
      */
     public function createConnection(array $settings)
     {
-        $dsn = 'pgsql:host='.$settings['hostname'].';dbname='.$settings['database'];
+        $dsn = 'pgsql:'.(($settings['hostname'] != '') ? 'host='.$settings['hostname'].';' : '').'dbname='.$settings['database'];
```
Comment (fguillot) [inline lib/PicoDb/Driver/Postgres.php]: You should set the hostname optional (line 23) and test for the existence of the key. In this way, if there is no hostname, the library will use unix sockets.

**Comment by yann-soubeyrand** [comment]
I updated my patch according to your review.

File: lib/PicoDb/Driver/Postgres.php [hunk 2]
```diff
@@ -42,12 +41,18 @@ class Postgres extends Base
...
     public function createConnection(array $settings)
     {
-        $dsn = 'pgsql:host='.$settings['hostname'].';dbname='.$settings['database'];
+        $dsn = 'pgsql:';
+
+        if (isset($settings['hostname'])) {
+            $dsn .= 'host='.$settings['hostname'].';';
+        }
 
         if (! empty($settings['port'])) {
```
Comment (yann-soubeyrand) [inline lib/PicoDb/Driver/Postgres.php]: What's the reason for using the empty() function here? Should I have used it to test the hostname setting too? Or isset() could be used for the port setting?

**Comment by bendem** [comment]
this was fixed in #23

**Pull request closed without merging**